marshal.go

```diff
@@ -57,6 +57,10 @@ func Marshal(info TypeInfo, value interface{}) ([]byte, error) {
 		}
 	}
 
+	if _, ok := value.(UnsetColumn); ok {
```

Comment:
I can move it into the if statement below with the custom unmarshaller for readability/maintainability or alternatively it can exist in each of the Unmarshal functions.  (Some data types don't support unset, tuple, map and a couple others)

It wouldn't fit nicely into the switch statement below because those are the Type info that relates to the row's types.  So a column wouldn't be of type Uset.  I'll fire up a branch with the marshalling changes in their respective types, it just seemed gross at the time.